packages/terra-application-header-layout/tests/jest/ApplicationHeader.test.jsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+import ApplicationHeaderLayout from '../../src/ApplicationHeaderLayout';
+
+describe('ApplicationHeaderLayout', () => {
+  it('should render without optional props', () => {
+    const result = shallow(<ApplicationHeaderLayout />);
+    expect(result).toMatchSnapshot();
+  });
+
+  it('should render with provided props', () => {
```

Comment:
Should we add a couple tests with different combinations of providing extensions, navigation, logo, etc since there are if checks on how to render them?